airflow/www_rbac/api/experimental/endpoints.py

```diff
@@ -80,21 +92,33 @@ def trigger_dag(dag_id):
             response.status_code = 400
 
             return response
+    else:
+        # set this here so that we can log it
+        execution_date = timezone.utcnow()
 
     replace_microseconds = (execution_date is None)
     if 'replace_microseconds' in data:
         replace_microseconds = to_boolean(data['replace_microseconds'])
 
     try:
+        with create_session() as session:
+            session.add(models.Log(
+                'api_trigger_dag',
+                task_instance=None,
+                owner=_get_user_from_header(),
+                dag_id=dag_id,
+                execution_date=execution_date,
+                extra=str(list(data.items()))
+            ))
+            session.commit()
         dr = trigger.trigger_dag(dag_id, run_id, conf, execution_date, replace_microseconds)
```

Comment:
What does this function do when `execution_date` is None? Are we meaningfully changing the behavior here?

Reply:
nope it just gets set later down the code path. setting it here means we can use it in the log entry

https://github.com/Affirm/airflow/blob/1.10.9-affirm/airflow/api/common/experimental/trigger_dag.py#L56